### PR TITLE
GluonCV YoloV3 Darknet53 example training and inference with Neo

### PR DESCRIPTION
**Pull request by bpawlik**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bartek-aws** [comment]
This is an end-to-end example of GluonCV YoloV3 model training inside of Amazon SageMaker notebook and then compile the trained model using Neo runtime.
This demo demonstrates how to train and to host a darknet53 model on the Pascal VOC dataset using the YoloV3 algorithm. It also demonstrates how to optimize this trained model using Neo.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The format of the output can be represented as `[class_index, confidence_score, xmin, ymin, xmax, ymax]`. Typically, we don't consider low-confidence predictions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:495]: set "cell_type" to "code" in order to define the neo_visualize_detection() function

Re: the next code block needs this function

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+    "            if classes and len(classes) > cls_id:\n",
+    "                class_name = classes[cls_id]\n",
+    "            plt.gca().text(xmin, ymin-2,\n",
+    "                            '{:s} {:.3f}'.format(class_name, score),\n",
+    "                            bbox=dict(facecolor=colors[cls_id], alpha=0.5),\n",
+    "                                    fontsize=12, color='white')\n",
+    "        plt.show()"
+   ]
+  },
+  {
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:550]: Remove empty code block

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "This is an end-to-end example of GluonCV YoloV3 model training inside of Amazon SageMaker notebook and then compile the trained model using Neo runtime. In this demo, we will demonstrate how to train and to host a darknet53 model on the [Pascal VOC dataset](http://host.robots.ox.ac.uk/pascal/VOC/) using the YoloV3 algorithm. We will also demonstrate how to optimize this trained model using Neo.\n",
+    "\n",
+    "***This is notebook is for demostration purpose only. Please fine tuning the training parameters based on your own dataset.***"
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:29]: typo: "Please fine tune the...."

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+    "Download the Pascal VOC datasets from 2007 and 2012."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:99]: Nice to have: 
store extracted data into S3 bucket
check that file does not exist

Thus, folks who return to the notebook don't need download and extract the data again.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb (hunk 1 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:493]: Thanks, I missed that. Done.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb (hunk 2 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:548]: Thanks, done.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb (hunk 3 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb]: Good catch, done.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb (hunk 4 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:99]: I have modified this notebook cell as follows: check whether file exists, if not then download and extract. I didn't want to involve S3 to spare user incurring unnecessary costs.

**Comment by bartek-aws** [comment]
@ngluna Thank you so much for your comments! I have addressed all of them, could you please review again?

Thanks,
Bartek

**Comment by ngluna** [comment]
Hey! Thanks for taking the time to reply to my comments. 

For some reason I can't see your changes...this might an error on my part. Can you double-check you committed the changes?

**Comment by bartek-aws** [comment]
Hi! I squashed the commits and overwritten the changes. I'm not sure that I can revert the squash and do a separate commit, let me check.

Thanks,
Bartek

**Comment by ngluna** [comment]
Ahh, gotcha. No worries if not. It might be a bit of effort to do so. 
I can go ahead and approve it.

**Comment by bpawlik** [comment]
@ngluna I have found an amazing feature in github that let me compare any (even unrelated commits) :) Here are my changes from after your review:
https://github.com/awslabs/amazon-sagemaker-examples/compare/96518ea..eb6c9df

If you approve, please go ahead and merge.

Thanks,
Bartek

**Comment by bartek-aws** [comment]
Hi @ngluna 
Are you able to merge this pull request? Thanks in advance!
Bartek

**Comment by ngluna** [comment]
Hi @bartek-aws,

I tried a couple of different ways to see the updates on the notebook. Couldn't find them. Thought maybe it was an error on my end. Went ahead and merged. But it looks like your changes weren't committed after all?

Should we make a new PR or revert this one?

**Comment by bartek-aws** [comment]
Hi @ngluna 

All changes have been commited and you already approved them 1.5 month ago :) I made mistake by force-pushing my changes to this branch and therefore github was unable to parse them well in this discussion. You can take a look at this diff that shows changes after your initial comments: https://github.com/awslabs/amazon-sagemaker-examples/compare/96518ea..eb6c9df

Thanks,
Bartek

**Comment by ngluna** [comment]
Right. That's what I thought. But if you look at the newly merged notebook: https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb

Some mistakes (extra cell, code cell as markdown) are still there?

**Comment by bpawlik** [comment]
@ngluna You're right, empty cell and markdown somehow stayed there.. Apologies for that. I have created a new PR with these two fixes: https://github.com/awslabs/amazon-sagemaker-examples/pull/1582

**Pull request merged**